### PR TITLE
chore: add bug, data, and feature template

### PR DESCRIPTION
**Pull request by kasparm**
Initial template.

I only added "(Required)" to the first paragraph in the feature request template following our discussion.

Please propose alternative or approve.

Thank you.

### CODE REVIEW

File: ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#
+```
+
+### Expected outcome
+A description of the expected outcome from the code snippet.
+
+### Environment
+Please specify your platform and versions of the relevant libraries you are using:
+* Operating system:
+* PowerSimData revision (run `git rev-parse origin/HEAD`):
```
Comment (jenhagg) [inline ISSUE_TEMPLATE/bug_report.md]: Maybe change this to "git revision"?

File: ISSUE_TEMPLATE/data-intake.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+* Name: ***e.g.*** Form EIA-923
+* Author: ***e.g.*** EIA
+* Description: ***e.g.*** Electric power data -- monthly and annually -- on electricity generation, fuel consumption, fossil fuel stocks, and receipts at the power plant and prime mover level for 2016
+* Source: ***e.g.*** https://www.eia.gov
+* Exact source location: ***e.g.*** https://www.eia.gov/electricity/data/eia923/
+* Terms (if specified): ***e.g.*** https://www.eia.gov/about/copyrights_reuse.php where it is written that we may use and/or distribute any of their data, files, databases, reports, graphs, charts, and other information products that are on their website
+
+### Destination
+* Modifications to source files(s): ***e.g.*** None
+* Location: powersimdata/...
```
Comment (jenhagg) [inline ISSUE_TEMPLATE/data-intake.md]: We could replace `powersimdata/...` with something generic, like `YOUR_REPO/...`

File: ISSUE_TEMPLATE/feature_request.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+ # :rocket:
+
+- [ ] Is your feature request essential for your project?
+
+
+### Describe the workflow you want to enable (Required)
+A clear and concise description of what can be enhanced, e.g., "I wish I could do [...]"
+
+### Describe your proposed implementation
```
Comment (jenhagg) [inline ISSUE_TEMPLATE/feature_request.md]: Based on our discussion, I might emphasize that this is optional or context dependent (as opposed to something we would prefer but isn't strictly required).

Comment (kasparm) [reply ISSUE_TEMPLATE/feature_request.md]: I will remove "(Required)" in line 15 and add to line 18 ",if familiar with code base":
### Describe your proposed implementation, if familiar with code base

Would that work?

Comment (jenhagg) [reply ISSUE_TEMPLATE/feature_request.md]: Hmm, I might leave it as just "optional" or "if applicable" - even if someone is familiar with the code I'd say it's fine (and in some cases, expected) to omit implementation.

**Review by jenhagg — APPROVED** [review]

**Review by BainanXia — APPROVED** [review]

**Pull request merged**